src/core/profile-provider/resolve-map-ast.test.ts

```diff
@@ -0,0 +1,422 @@
+  });
+
+  it('throws when profile ids does not match - scope is different', async () => {
+    await expect(
+      resolveMapAst({
+        profileId: 'different-scope/test',
+        providerName,
+        variant,
+        fileSystem: mockFileSystem(
+          `path/to${EXTENSIONS.map.build}`,
+          ok(
+            JSON.stringify(
+              mockMapDocumentNode({
+                name: 'name',
+                provider: providerName,
+                variant,
+              })
+            )
+          )
+        ),
+        superJson: {
+          profiles: {
+            ['different-scope/test']: {
+              version: '1.0.0',
+              providers: {
+                [providerName]: {
+                  file: `path/to${EXTENSIONS.map.source}`,
+                  defaults: {},
+                },
+              },
+              defaults: {},
+              priority: [providerName],
+            },
+          },
+          providers: {},
+        },
+        config,
+      })
+    ).rejects.toThrow(
+      profileIdsDoNotMatchError('name', 'different-scope/test')
+    );
+  });
+
+  it('throws when profile ids does not match - name is different', async () => {
+    await expect(
+      resolveMapAst({
+        profileId: 'scope/different',
+        providerName,
+        variant,
+        fileSystem: mockFileSystem(
+          `path/to${EXTENSIONS.map.build}`,
+          ok(JSON.stringify(ast))
+        ),
+        superJson: {
+          profiles: {
+            ['scope/different']: {
+              version: '1.0.0',
+              providers: {
+                [providerName]: {
+                  file: `path/to${EXTENSIONS.map.source}`,
+                  defaults: {},
+                },
+              },
+              defaults: {},
+              priority: [providerName],
+            },
+          },
+          providers: {},
+        },
+        config,
+      })
+    ).rejects.toThrow(
+      profileIdsDoNotMatchError('scope/name', 'scope/different')
+    );
+  });
+
+  it('throws when provider names does not match', async () => {
+    await expect(
+      resolveMapAst({
+        profileId,
+        providerName: 'different',
+        variant,
+        fileSystem: mockFileSystem(
+          `path/to${EXTENSIONS.map.build}`,
+          ok(JSON.stringify(ast))
+        ),
+        superJson: {
+          profiles: {
+            [profileId]: {
+              version: '1.0.0',
+              providers: {
+                ['different']: {
+                  file: `path/to${EXTENSIONS.map.source}`,
+                  defaults: {},
+                },
+              },
+              defaults: {},
+              priority: ['different'],
+            },
+          },
+          providers: {},
+        },
+        config,
+      })
+    ).rejects.toThrow(providersDoNotMatchError('provider', 'different', 'map'));
+  });
+
+  it('return map ast when path to source is passed', async () => {
```

Comment:
Can you also add here that it expects that ast is next to source, I thought that the next test is duplicate. 😄

Reply:
Martin has a good point

Reply:
nitpick: 
```suggestion
  it('returns map ast when path to source is passed', async () => {
```